core/src/test/java/org/infinispan/statetransfer/StateTransferPessimisticTest.java

```diff
@@ -106,4 +110,20 @@ public int checkKey(Object key, Cache... caches) {
       }
       return c;
    }
+
+   protected void waitUntilDataContainerWillBeIteratedOn(final Cache<?, ?> cache, final CyclicBarrier barrier) {
```

Comment:
Looks more like a `blockDataContainerIteration()` to me ;)

Also, doesn't it block the cache clear during teardown?